packages/@aws-cdk/aws-batch/lib/ecs-container-definition.ts

```diff
@@ -0,0 +1,949 @@
+  }
+}
+
+/**
+ * Options for configuring an EfsVolume
+ */
+export interface EfsVolumeOptions extends EcsVolumeOptions {
+  /**
+   * The EFS File System that supports this volume
+   */
+  readonly fileSystem: IFileSystem;
+
+  /**
+   * The directory within the Amazon EFS file system to mount as the root directory inside the host.
+   * If this parameter is omitted, the root of the Amazon EFS volume is used instead.
+   * Specifying `/` has the same effect as omitting this parameter.
+   * The maximum length is 4,096 characters.
+   *
+   * @default - root of the EFS File System
+   */
+  readonly rootDirectory?: string;
+
+  /**
+   * Enables encryption for Amazon EFS data in transit between the Amazon ECS host and the Amazon EFS server
+   *
+   * @see https://docs.aws.amazon.com/efs/latest/ug/encryption-in-transit.html
+   *
+   * @default false
+   */
+  readonly enableTransitEncryption?: boolean;
+
+  /**
+   * The port to use when sending encrypted data between the Amazon ECS host and the Amazon EFS server.
+   * The value must be between 0 and 65,535.
+   *
+   * @see https://docs.aws.amazon.com/efs/latest/ug/efs-mount-helper.html
+   *
+   * @default - chosen by the EFS Mount Helper
+   */
+  readonly transitEncryptionPort?: number;
+
+  /**
+   * The Amazon EFS access point ID to use.
+   * If an access point is specified, `rootDirectory` must either be omitted or set to `/`
+   * which enforces the path set on the EFS access point.
+   * If an access point is used, `enableTransitEncryption` must be `true`.
+   *
+   * @see https://docs.aws.amazon.com/efs/latest/ug/efs-access-points.html
+   *
+   * @default - no accessPointId
+   */
+  readonly accessPointId?: string;
+
+  /**
+   * Whether or not to use the AWS Batch job IAM role defined in a job definition when mounting the Amazon EFS file system.
+   * If specified, `enableTransitEncryption` must be `true`.
+   *
+   * @see https://docs.aws.amazon.com/batch/latest/userguide/efs-volumes.html#efs-volume-accesspoints
+   *
+   * @default false
+   */
+  readonly useJobDefinitionRole?: boolean;
+}
+
+/**
+ * A Volume that uses an AWS Elastic File System (EFS); this volume can grow and shrink as needed
+ */
+export class EfsVolume extends EcsVolume {
+  /**
+   * Returns true if x is an EfsVolume, false otherwise
+   */
+  public static isEfsVolume(x: any) : x is EfsVolume {
+    return x !== null && typeof(x) === 'object' && EFS_VOLUME_SYMBOL in x;
+  }
+
+  /**
+   * The EFS File System that supports this volume
+   */
+  public readonly fileSystem: IFileSystem;
+
+  /**
+   * The directory within the Amazon EFS file system to mount as the root directory inside the host.
+   * If this parameter is omitted, the root of the Amazon EFS volume is used instead.
+   * Specifying `/` has the same effect as omitting this parameter.
+   * The maximum length is 4,096 characters.
+   *
+   * @default - root of the EFS File System
+   */
+  public readonly rootDirectory?: string;
```

Comment:
For a simple integration class, not sure we need all these. I suppose it doesn't hurt either now that you've written all of it already, but people aren't usually going to be interacting with this object.